src/creator/wallet-creator.service.ts

```diff
@@ -0,0 +1,37 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from 'nestjs-prisma';
+import { WalletCreatorStats } from './types/my-stats';
+
+@Injectable()
+export class WalletCreatorService {
+  constructor(private readonly prisma: PrismaService) {}
+  async follow(walletAddress: string, creatorSlug: string): Promise<boolean> {
+    let walletCreator = await this.prisma.walletCreator.findUnique({
+      where: {
+        creatorSlug_walletAddress: { walletAddress, creatorSlug },
+      },
+    });
+
+    walletCreator = await this.prisma.walletCreator.upsert({
+      where: { creatorSlug_walletAddress: { walletAddress, creatorSlug } },
+      create: { creatorSlug, walletAddress, isFollowing: true },
+      update: { isFollowing: !walletCreator?.isFollowing },
+    });
+    return !!walletCreator;
+  }
```

Comment:
Do you think this should be done similar to what we did with `toggleState` on comic issues and comics?

Reply:
Sure thing. It's the name that confused me... I thought you forgot to do the "unfollow" because we only had the "follow" function! 🙃 

Could you at least rename it to `toggleFollow`, `toggleFollowStatus` or something along those lines so my french fry sized brain can understand what it does just from it's name! 🍟

Reply:
togleFollow sounds like a good name